### PR TITLE
Place floating point routines into a dedicated header

### PR DESCRIPTION
**Pull request by 2011eric**

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Remove `src/soft_float.c` since it is needed at the moment.

File: src/soft_float.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    FFLAG_DIV_BY_ZERO = 0b00000000000000000000000000001000,
+    FFLAG_OVERFLOW    = 0b00000000000000000000000000000100,
+    FFLAG_UNDERFLOW   = 0b00000000000000000000000000000010,
+    FFLAG_INEXACT     = 0b00000000000000000000000000000001,
+    //                    ....xxxx....xxxx....xxxx....xxxx
+    RV_NAN            = 0b01111111110000000000000000000000
+};
+
+
+/* clang-format on */
```
Comment (jserv) [inline src/soft_float.h]: Move `/* clang-format on */` nearby the above `enum`.

File: src/emulate.c [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 #if RV32_HAS(EXT_F)
 #include <math.h>
+#include "soft_float.h"
```
Comment (jserv) [inline src/emulate.c:16]: Add blank line before and after `soft_float.h`.

**Comment by 2011eric** [comment]
Should I also modify the commit title?

**Comment by jserv** [comment]
> Should I also modify the commit title?

Go ahead.

**Pull request merged**